### PR TITLE
Preview of German

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**

### CODE REVIEW

File: packages/pyright-internal/src/localization/simplified.nls.de.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
+{
+    "Diagnostic": {
+        "argAssignment": "Argument stimmt nicht mit Parametertyp überein",
+        "argAssignmentFunction": "Argument stimmt nicht mit dem Parametertyp in Funktion \"{functionName} \" überein",
+        "argAssignmentParam": "Argument stimmt nicht mit Parametertyp für Parameter \"{paramName} \" überein",
```
Comment (microbit-robert) [inline packages/pyright-internal/src/localization/simplified.nls.de.json]: `functionName` and `paramName` have an extra space after the curly brace in both cases.

File: packages/pyright-internal/src/localization/simplified.nls.de.json [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        "expectedEqualityOperator": "Hier wird ein Gleichheitsoperator erwartet, meinten Sie \"==\"?",
+        "expectedExpr": "Ausdruck fehlt",
+        "expectedFunctionName": "Funktionsname nach \"def\" fehlt.",
+        "expectedIdentifier": "Bezeichner fehlt",
+        "expectedImport": "„import“ fehlt",
+        "expectedImportAlias": "Fehlendes Symbol nach \"as\"",
+        "expectedImportSymbols": "Nach dem Import wird ein oder mehrere Symbolnamen erwartet",
+        "expectedIn": "„in“ fehlt.",
+        "expectedIndentedBlock": "Einrückung fehlt",
+        "expectedInExpr": "Fehlender Ausdruck nach \"in\"",
```
Comment (microbit-robert) [inline packages/pyright-internal/src/localization/simplified.nls.de.json]: There's a mix of quote types. Some are " and some are „

There doesn't seem to be any logic to this. See `in` in `expectedIn` and `expectedInExpr`.

File: packages/pyright-internal/src/localization/simplified.nls.de.json [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        "inconsistentIndent": "Einrückung stimmt nicht mit der vorherigen Zeile überein",
+        "inconsistentTabs": "Inkonsistente Verwendung von Tabs und Leerzeichen für die Einrückung",
+        "instanceMethodSelfParam": "Instanzmethoden benötigen einen \"self\"-Parameter",
+        "invalidIdentifierChar": "Ungültiges Zeichen im Bezeichner",
+        "invalidTokenChars": "Ungültiges Zeichen \"{text}\" im Token",
+        "memberAccess": "Kann nicht auf Mitglied „{name}“ für Typ „{type}“ zugreifen.",
+        "memberDelete": "Kann Mitglied \"{name}\"für Typ \"{type}\" nicht löschen",
+        "memberSet": "Kann Mitglied \"{name}\" nicht für Typ \"{type}\" zuweisen",
+        "moduleNotCallable": "Modul ist nicht abrufbar.",
+        "moduleUnknownMember": "„{name}“ ist kein bekanntes Mitglied des Moduls „{module}“.",
```
Comment (microbit-robert) [inline packages/pyright-internal/src/localization/simplified.nls.de.json]: There's a few unexpected full stops, mostly towards the end of the file.

File: packages/pyright-internal/src/localization/simplified.nls.de.json [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        "expectedModuleName": "Modulname fehlt",
+        "expectedNamedParameter": "Schlüsselwortparameter muss \"*\" folgen",
+        "expectedNewlineOrSemicolon": "Unerwartete zusätzliche Inhalte\nEs wird eine Anweisung pro Zeile erwartet",
+        "expectedOpenParen": "Öffnende runde Klammer „)“ fehlt.",
+        "expectedParamName": "Fehlender Parametername",
+        "expectedSliceIndex": "Fehlender Index oder Slice",
+        "expectedYieldExpr": "Ein Ausdruck wird in der yield-Anweisung erwartet",
+        "globalReassignment": "\"{name}\" wird vor der globalen Deklaration zugewiesen",
+        "globalRedefinition": "\"{name}\" wurde bereits als global deklariert",
+        "importResolveFailure": "Modul „{importName}“ konnte nicht gefunden werden.",
```
Comment (microbit-robert) [inline packages/pyright-internal/src/localization/simplified.nls.de.json]: The space after Modul is `U+00a0`. Flagged in VSCode, presumably no impact.

File: packages/pyright-internal/src/localization/simplified.nls.de.json (hunk 1 above)
Comment (microbit-matt-hillsdon) [reply packages/pyright-internal/src/localization/simplified.nls.de.json]: Thanks - I'll fix this and do another update.

File: packages/pyright-internal/src/localization/simplified.nls.de.json (hunk 2 above)
Comment (microbit-matt-hillsdon) [reply packages/pyright-internal/src/localization/simplified.nls.de.json]: I'll raise this with Emma to feed back to the translator/proofreader.

File: packages/pyright-internal/src/localization/simplified.nls.de.json (hunk 3 above)
Comment (microbit-matt-hillsdon) [reply packages/pyright-internal/src/localization/simplified.nls.de.json]: I may as well fix these

File: packages/pyright-internal/src/localization/simplified.nls.de.json (hunk 4 above)
Comment (microbit-matt-hillsdon) [reply packages/pyright-internal/src/localization/simplified.nls.de.json]: I may as well fix these

File: packages/pyright-internal/src/localization/simplified.nls.de.json (hunk 3 above)
Comment (microbit-matt-hillsdon) [reply packages/pyright-internal/src/localization/simplified.nls.de.json]: Raised with Emma as they've been introduced recently

**Comment by microbit-matt-hillsdon** [comment]
Happy with this as a preview now. Merging.

**Pull request merged**